src/themes/default/components/core/blocks/Microcart/Product.vue

```diff
@@ -224,6 +247,56 @@ export default {
     },
     updateQuantity (quantity) {
       this.$store.dispatch('cart/updateQuantity', { product: this.product, qty: quantity })
+    },
+    async getQuantity (product) {
+      if (this.isStockInfoLoading) return // stock info is already loading
+      this.isStockInfoLoading = true
+      try {
+        const validProduct = product || this.product
+        const res = await this.$store.dispatch('stock/check', {
+          product: validProduct,
+          qty: this.productQty
+        })
+        return res.qty
+      } finally {
+        this.isStockInfoLoading = false
+      }
+    },
+    getQuantityError (error) {
+      this.quantityError = error
+    },
+    async changeEditModeFilter (filter) {
+      const editedProduct = this.getEditedProduct(filter)
+      const maxQuantity = await this.getQuantity(editedProduct)
+      if (!maxQuantity) {
+        this.$store.dispatch('notification/spawnNotification', {
+          type: 'error',
+          message: this.$t(
+            'The product is out of stock and cannot be added to the cart!'
+          ),
+          action1: { label: this.$t('OK') }
+        })
+      } else if (maxQuantity < this.productQty) {
+        this.$store.dispatch('notification/spawnNotification', {
+          type: 'error',
+          message: this.$t('Only') + ` ${maxQuantity} ` + this.$t('products of this type are available!'),
```

Comment:
```suggestion
          message: this.$t('Only {maxQuantity} products of this type are available!', { maxQuantity }),
```
and please adjust translations for that

Reply:
done